### PR TITLE
factory: Add new factory option template_path

### PR DESCRIPTION
**Pull request by teawater**
Add new factory option template_path that specifies the path of template.

Fixes: #1549

Signed-off-by: Hui Zhu <teawater@hyper.sh>

### CODE REVIEW

File: cli/config/configuration-qemu.toml.in [hunk 1]
```diff
@@ -219,6 +219,11 @@ enable_iothreads = @DEFENABLEIOTHREADS@
 # Default false
 #enable_template = true
 
+# Specifies the path of template.
+#
+# Default "/run/vc/vm/template"
+#template_path = "/run/vc/vm/template"
```
Comment (jodh-intel) [inline cli/config/configuration-qemu.toml.in]: Do you also need to update `configuration-fc.toml.in`?

Comment (teawater) [reply cli/config/configuration-qemu.toml.in]: > Do you also need to update `configuration-fc.toml.in`?

@jodh-intel fc cannot support template now.

**Pull request merged**